lib/rex/post/meterpreter/extensions/stdapi/railgun/def/windows/def_iphlpapi.rb

```diff
@@ -43,27 +43,27 @@ def self.create_library(constant_manager, library_path = 'iphlpapi')
 
     dll.add_function('GetBestInterface', 'DWORD',[
       ["DWORD","dwDestAddr","in"],
-      ["PDWORD","pdwBestIfIndex","inout"],
+      ["PDWORD","pdwBestIfIndex","out"],
       ])
 
     dll.add_function('GetBestInterfaceEx', 'DWORD',[
       ["PBLOB","pDestAddr","in"],
-      ["PDWORD","pdwBestIfIndex","inout"],
+      ["PDWORD","pdwBestIfIndex","out"],
       ])
 
     dll.add_function('GetFriendlyIfIndex', 'DWORD',[
       ["DWORD","IfIndex","in"],
       ])
 
     dll.add_function('GetNumberOfInterfaces', 'DWORD',[
-      ["PDWORD","pdwNumIf","inout"],
+      ["PDWORD","pdwNumIf","out"],
       ])
 
     dll.add_function('GetRTTAndHopCount', 'BOOL',[
       ["DWORD","DestIpAddress","in"],
-      ["PDWORD","HopCount","inout"],
+      ["PDWORD","HopCount","out"],
       ["DWORD","MaxHops","in"],
-      ["PDWORD","RTT","inout"],
+      ["PDWORD","RTT","out"],
       ])
```

Comment:
Would you mind adding PULONG as an alias to PDWORD in [library.rb](https://github.com/rapid7/metasploit-framework/blob/master/lib/rex/post/meterpreter/extensions/stdapi/railgun/library.rb#L46) so we can define this correctly and not come back to them?

```suggestion
    dll.add_function('GetRTTAndHopCount', 'BOOL',[
      ["DWORD","DestIpAddress","in"],
      ["PULONG","HopCount","out"],
      ["DWORD","MaxHops","in"],
      ["PULONG","RTT","out"],
      ])
```

With that in place, the definition will be a 1:1 [match with the docs.](https://learn.microsoft.com/en-us/windows/win32/api/iphlpapi/nf-iphlpapi-getrttandhopcount)

Reply:
Sounds good I'll add an update for this in now 👍

Reply:
Should be fixed with https://github.com/rapid7/metasploit-framework/pull/17561/commits/b5a83ffd0ffed9a9a39c4986398b00307216451f